File: src/main/java/com/sedmelluq/discord/lavaplayer/tools/BigRandom.java
package com.sedmelluq.discord.lavaplayer.tools;

import java.math.BigInteger;
import java.util.Random;

public final class BigRandom extends Random {

  public static void main(final String[] args) {
    final BigRandom random = new BigRandom();
    final BigInteger bigInt = random.nextBigInt(2048);
    System.out.println(bigInt);
  }

  public BigInteger nextBigInt(int bits) {
    if (bits < 32) {
      return BigInteger.valueOf(next(31));
    }
    BigInteger value = BigInteger.ZERO;
    int index = 0;
    while (bits >= 32) {
      bits -= 32;
      value = value.add(BigInteger.valueOf(next(32)).shiftLeft((index++) * 32));
    }
    if (bits > 0)
      value = value.add(BigInteger.valueOf(next(bits)).shiftLeft(index * 32));
    return value;
  }
}

File: extensions/ip-rotator/src/main/java/lavaplayer/extensions/iprotator/tools/BigRandom.kt
package lavaplayer.extensions.iprotator.tools

import java.math.BigInteger
import java.util.*

class BigRandom : Random() {

    fun nextBigInt(size: Int): BigInteger {
        if (size < 32) {
            return next(31).toBigInteger();
        }

        var bits = size
        var value = BigInteger.ZERO
        var index = 0
        while (bits >= 32) {
            bits -= 32;
            value = value.add(next(32).toBigInteger() shl (index++) * 32);
        }

        if (bits > 0) {
            value = value.add(next(bits).toBigInteger() shl index * 32);
        }

        return value;
    }

}
